torch/_dynamo/symbolic_convert.py

```diff
@@ -277,6 +277,21 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
             if truth_fn(value):
                 push and self.push(value)
                 self.jump(inst)
+        elif isinstance(value, UserDefinedObjectVariable):
+            if "__bool__" in value.value.__class__.__dict__:
+                result = self.inline_user_function_return(
+                    UserFunctionVariable(getattr(value.value, "__bool__").__func__),
+                    [value],
+                    {},
+                )
+                assert isinstance(result, ConstantVariable)
```

Comment:
we should call `unimplemented()` to graph break cleanly here.  Might need to also checkpoint/rollback.

Reply:
Python ```__bool__``` function always return bool.